controllers/provisioners/ekscloudformation/helpers.go

```diff
@@ -47,3 +49,38 @@ func getConfigMap(k kubernetes.Interface, namespace string, name string, options
 	}
 	return cm, nil
 }
+
+func addAnnotation(u *unstructured.Unstructured, key, value string) {
+	annotations := u.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+	annotations[key] = value
+	u.SetAnnotations(annotations)
+}
+
+func hasAnnotation(u *unstructured.Unstructured, key, value string) bool {
+	annotations := u.GetAnnotations()
+	if val, ok := annotations[key]; ok {
+		if val == value {
+			return true
+		}
+	}
+	return false
+}
+
+func getUnstructuredPath(u *unstructured.Unstructured, jsonPath string) (string, error) {
+	splitFunction := func(c rune) bool {
+		return c == '.'
+	}
+	statusPath := strings.FieldsFunc(jsonPath, splitFunction)
+
+	value, ok, err := unstructured.NestedString(u.UnstructuredContent(), statusPath...)
+	if err != nil {
+		return "", err
+	}
+	if ok {
+		return value, nil
```

Comment:
What's in `value` if `ok` is false? You could probably just return value here if it's already empty.
Another improvement for later is to possibly return here `interface{}` type and reflect.Type the call to `Nested{Type}` function.